src/sssom/util.py

```diff
@@ -515,11 +514,13 @@ def get_row_based_on_hierarchy(df: pd.DataFrame):
 
     :param df: Dataframe containing multiple predicates for same subject and object.
     :return: Dataframe with a single row which ranks higher in the hierarchy.
+    :raises KeyError: if no rows are available
     """
     for pred in PREDICATE_LIST:
         hierarchical_df = df[df[PREDICATE_ID] == pred]
         if not hierarchical_df.empty:
             return hierarchical_df
+    raise KeyError
```

Comment:
What is the motivation behind this change?

Reply:
The code that uses this function always assumes it returns something. before, there was an implicit return of None at the end. If for whatever reason none of the predicates in the predicate list are available, this code should explicitly break, rather than ending up with a type error when trying to concat a None onto the dataframe that gets returned. This is why strict typing is helpful!

Reply:
Ok, thank you!

Reply:
I don't really understand what this method is doing. I guess your change is independent of that, and makes sense